packages/sdk/src/oracle/services/skinnyOptimisticOracle.ts

```diff
@@ -0,0 +1,159 @@
+import assert from "assert";
+import { skinnyOptimisticOracle as optimisticOracle } from "../../clients";
+import { BigNumberish, Provider, Signer, TransactionResponse, Log, TransactionReceipt } from "../types/ethers";
+import type { OracleInterface, RequestKey, OracleProps, Request } from "../types/interfaces";
+import { requestId, insertOrderedAscending, eventKey, isUnique, getAddress } from "../utils";
+
+import {
+  RequestPrice,
+  ProposePrice,
+  DisputePrice,
+  Settle,
+  SolidityRequest,
+} from "../../clients/skinnyOptimisticOracle";
+
+export type OptimisticOracleEvent = RequestPrice | ProposePrice | DisputePrice | Settle;
+
+export type { RequestPrice, ProposePrice, DisputePrice, Settle };
+
+export function getOptimisticOracleAddress(chainId: number): string {
+  switch (chainId.toString()) {
+    case "1":
+      return getAddress("0xeE3Afe347D5C74317041E2618C49534dAf887c24");
+    case "4":
+      return getAddress("0xAbE04Ace666294aefD65F991d78CE9F9218aFC67");
+    case "42":
+      return getAddress("0xB1d3A89333BBC3F5e98A991d6d4C1910802986BC");
+    case "100":
+      return getAddress("0xAa04b5D40574Fb8C001249B24d1c6B35a207F0bD");
+    default:
+      throw new Error(`No address found for deployment Skinny Optimistic Oracle on chainId ${chainId}`);
+  }
+}
+
+function validateSolidityRequest(request: Request): SolidityRequest {
+  assert(request.proposer, "Missing proposer");
+  assert(request.disputer, "Missing disputer");
+  assert(request.currency, "Missing currency");
+  assert(request.settled, "Missing settled");
+  assert(request.proposedPrice, "Missing proposedPrice");
+  assert(request.resolvedPrice, "Missing resolvedPrice");
+  assert(request.expirationTime, "Missing expirationTime");
+  assert(request.reward, "Missing reward");
+  assert(request.finalFee, "Missing finalFee");
+  assert(request.bond, "Missing bond");
+  assert(request.customLiveness, "Missing customLiveness");
+  return {
+    proposer: request.proposer,
+    disputer: request.disputer,
+    currency: request.currency,
+    settled: request.settled,
+    proposedPrice: request.proposedPrice,
+    resolvedPrice: request.resolvedPrice,
+    expirationTime: request.expirationTime,
+    reward: request.reward,
+    finalFee: request.finalFee,
+    bond: request.bond,
+    customLiveness: request.customLiveness,
+  };
+}
+
+export class OptimisticOracle implements OracleInterface {
```

Comment:
Something I find a little confusing about this file is that the class and the getOptimisticOracleAddress are named the same as the other file, but they apply to the skinny optimistic oracle, not the optimistic oracle. Do these need to have the same names? It might improve readability if the skinny name was at least reflected in the names of the exported classes and functions.